drivers/saul/pwm_saul.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_saul
+ * @{
+ *
+ * @file
+ * @brief       SAUL wrapper for direct access to PWM pins
+ *
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "saul/periph.h"
+#include "phydat.h"
+#include "periph/pwm.h"
+
+static int set_pwm(const void *dev, phydat_t *value)
+{
+    const saul_pwm_params_t *params = *((const saul_pwm_params_t **)dev);
+    uint32_t freq;
+    uint16_t res;
+
+    if (value->val[1] > 0) {
+        freq = value->val[1];
+    }
+    else {
+        freq = params->freq;
+    }
+
+    if (value->val[2] > 0) {
+        res = value->val[2];
+    }
+    else {
+        res = params->res;
+    }
+
+    pwm_init(params->line, params->mode, freq, res);
```

Comment:
as this is a set functin (and the duty cycle being the set value), I don't see the init function being part of this. 

If the rationale here is to actually be able to adapt the PWM parameters using the SAUL interface, I think we need to re-think this approach and figure out something more elegant -> we might even have to re-visit the peripheral PWM interface, as it is in the current form not very suitable for on-the-fly adaption of the PWM parameters...